src/arch/InputHandler/InputHandler_Linux_Event.cpp

```diff
@@ -143,11 +288,17 @@ bool EventDevice::Open( RString sFile, InputDevice dev )
 	if( ioctl(m_iFD, EVIOCGBIT(EV_ABS, sizeof(iABSMask)), iABSMask) < 0 )
 		LOG->Warn( "ioctl(EVIOCGBIT(EV_ABS)): %s", strerror(errno) );
 
-	if( !BitIsSet(iABSMask, ABS_X) && !BitIsSet(iABSMask, ABS_THROTTLE) && !BitIsSet(iABSMask, ABS_WHEEL) )
+	if( BitIsSet(iABSMask, ABS_X) || BitIsSet(iABSMask, ABS_THROTTLE) || BitIsSet(iABSMask, ABS_WHEEL) )
```

Comment:
What would happen here if the device is not a keyboard, yet gets mapped as one?

Reply:
Essentially nothing.
The device will appear in the devices list (map/test input screen), which can have a large list including odd things like hardware buttons, hard drive impact sensors etc.
However unless the device actually sends a keypress/release event it will be ignored by the event driver. Non-keyboard devices shouldn't trigger any of the key codes, so would be ignored in LinuxKeyCodeToDeviceButton.

At this level the best we can do is check whether the device has axis, or whether its name contains the word 'keyboard' or 'IPAC' as in my first hack (All of mine do, but the name is a free text field).

One possibility would be to map ipac/other names, plus keyboard all the time, and only allow additional devices if listed in Preferences.ini.
The downside to this is that if no keyboard is handled by either the Event or X11 driver it's impossible to navigate the input mapping screen to setup other controllers.

The last thing I want to do is break keyboard input, and my limited testing can't possibly check all scenarios for this.